FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -212,7 +227,126 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
+        {
+            if (_inputData.JobVersion <= JobVersion.V4)
+            {
+                progress.Report(JsonConvert.SerializeObject(GenerateJobResultFromStatusAsync()));
+            }
+
+            if(_inputData.JobVersion >= JobVersion.V4)
+            {
+                // Update job status on meta table for job version larger than V4.
+                _jobStatusEntity.JobStatus = JsonConvert.SerializeObject(_jobStatus);
+
+                if (!await _metadataStore.TryUpdateEntityAsync(_jobStatusEntity))
+                {
+                    _logger.LogInformation("Update orchestrator job status failed in job {0}.", _jobInfo.Id);
+                    throw new SynapsePipelineInternalException($"Update orchestrator job status failed in job  {_jobInfo.Id}.");
+                }
+
+                _jobStatusEntity = await _metadataStore.GetOrchestratorJobStatusAsync(_jobInfo.QueueType, _jobInfo.GroupId, _jobInfo.Id, cancellationToken);
+            }
+        }
+
         private async IAsyncEnumerable<FhirToDataLakeProcessingJobInputData> GetInputsAsyncForSystem([EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            // Resume the submitting jobs.
+            if (_jobStatus.SubmittingProcessingJob != null)
+            {
+                foreach (var job in _jobStatus.SubmittingProcessingJob)
+                {
+                    yield return JsonConvert.DeserializeObject<FhirToDataLakeProcessingJobInputData>(job);
+                }
+            }
+
+            IAsyncEnumerable<FhirToDataLakeSplitProcessingJobInfo> processingJobs = _jobSplitter.SplitJobAsync(_inputData.DataStartTime, _inputData.DataEndTime, _jobStatus.SubmittedResourceTimestamps, cancellationToken);
+            await foreach (FhirToDataLakeSplitProcessingJobInfo processingJob in processingJobs.WithCancellation(cancellationToken))
+            {
+                if (processingJob.ResourceCount == 0)
+                {
+                    // If resource count is 0, skip the job and update the submitted timestamp.
+                    foreach (FhirToDataLakeSplitSubJobInfo subJob in processingJob.SubJobInfos)
+                    {
+                        _jobStatus.SubmittedResourceTimestamps[subJob.ResourceType] = subJob.TimeRange.DataEndTime;
+                    }
+
+                    continue;
+                }
+
+                yield return new FhirToDataLakeProcessingJobInputData
+                {
+                    JobType = JobType.Processing,
+                    JobVersion = _inputData.JobVersion,
+                    ProcessingJobSequenceId = _jobStatus.CreatedJobCount,
+                    TriggerSequenceId = _inputData.TriggerSequenceId,
+                    Since = _inputData.Since,
+                    SplitParameters = processingJob.SubJobInfos.ToDictionary(x => x.ResourceType, x => x.TimeRange),
```

Comment:
Why not directly reference a list of subjobinfos here? Since we might have more information for subjobs in the future

Reply:
Updated it and referenced the processingJobJnfo here.